src/gtk/toga_gtk/widgets/numberinput.py

```diff
@@ -5,37 +5,59 @@
 
 class NumberInput(Widget):
     def create(self):
-        adjustment = Gtk.Adjustment(0, self.interface.min_value,
-                                    self.interface.max_value,
-                                    self.interface.step, 10, 0)
-
         self.native = Gtk.SpinButton()
-        self.native.set_adjustment(adjustment)
         self.native.set_numeric(True)
         self.native.interface = self.interface
 
         self.rehint()
 
+    def set_adjustment_value(self):
+        adjustment = Gtk.Adjustment(0, self.interface.min_value,
+                                        self.interface.max_value,
+                                        self.interface.step, 10, 0)
+        self.native.set_adjustment(adjustment)
+
     def set_readonly(self, value):
-        self.native.editable = not value
+        self.native.set_property('editable', not value)
 
     def set_step(self, step):
-        self.interface.factory.not_implemented('NumberInput.set_step()')
+        if step:
+            self.native.set_increments(step,step)
 
     def set_min_value(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_min_value()')
+        #self.native.set_range(min=value)
+        if value:
+            try:
+                self.native.set_range(value,self.interface.max_value)
+            except (AttributeError):
+                pass
 
     def set_max_value(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_max_value()')
+        # self.native.set_range(max=value)
+        if value:
+            try:
+                self.native.set_range(self.interface.min_value,value)
+            except (AttributeError):
+                pass
 
     def set_value(self, value):
         self.native.set_value(value)
 
     def set_alignment(self, value):
-        self.interface.factory.not_implemented('NumberInput.set_alignment()')
+        if value == 'left':
```

Comment:
LEFT and RIGHT are symbolic constants which should be used in preference to 'left' and 'right'

Reply:
Are they by convention and need to be defined?
```
  File "/home/yugesh/Documents/beeware/toga/src/gtk/toga_gtk/widgets/numberinput.py", line 43, in set_alignment
    if value == LEFT:
NameError: name 'LEFT' is not defined

```
Or do they exist and need to be imported?

Reply:
Well there's nothing magic going on - if the symbol isn't imported, it can't be used. So yes - it needs to be imported.